lib/git.rb

```diff
@@ -127,6 +133,64 @@ def clone(remote, local, options = {})
       RubyGit.clone(remote, local, options)
     end
 
+    class RJGitSSHConfigCallback
+      include TransportConfigCallback
+      def initialize(options = {})
+        @sshSessionFactory = Class.new(JschConfigSessionFactory) {
+          def initialize(options)
+            super()
+            @private_key_file = options[:private_key_file]
+            @private_key_passphrase = options[:private_key_passphrase]
+            @username = options[:username]
+            @password = options[:password]
+            @known_hosts_file = options[:known_hosts_file]
+          end
+
+          def configure(host, session)
+            session.setUserName(@username) if @username
+            session.setPassword(@password) if @password
+          end
+
+          def createDefaultJSch(fs)
+            default_j_sch = super(fs)
+            if @private_key_file
+              default_j_sch.removeAllIdentity()
+              if @private_key_passphrase
+                default_j_sch.addIdentity(@private_key_file, @private_key_passphrase)
+              else
+                default_j_sch.addIdentity(@private_key_file)
+              end
+            end
+            if @known_hosts_file
+              default_j_sch.setKnownHosts(@known_hosts_file)
+            end
+
+            return default_j_sch
+          end
+        }.new(options)
+      end
+
+      def configure(ssh_transport)
+        ssh_transport.setSshSessionFactory(@sshSessionFactory)
+      end
+    end
+
+    def self.set_command_transport(command, remote, options = {})
+      valid_transports = SSH_TRANSPORTS + HTTP_TRANSPORTS + FILE_TRANSPORTS
+      uri = nil
+      begin
+        uri = URI.parse(remote) if remote
+      rescue URI::InvalidURIError
+      end
+      transport_protocol = uri.scheme if uri && (valid_transports.include? uri.scheme)
+
+      if (SSH_TRANSPORTS.include? transport_protocol.to_s) || (options[:transport_protocol] == :ssh) || options[:private_key_file]
+        command.set_transport_config_callback(RJGitSSHConfigCallback.new(options))
+      elsif (HTTP_TRANSPORTS.include? transport_protocol.to_s) || options[:username]
```

Comment:
The II allows for the case where you have a username/password for a local repo.
Here is a test case you have for this:
https://github.com/repotag/rjgit/blob/master/spec/git_spec.rb#L270